### PR TITLE
(SIMP-6667) Update Password Complexity FAQ

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Updated the password complexity FAQ to note the new settings as well
  as give some examples for items that routinely confuse users.
* Fix a link and some spelling issues in the Client Management section
  of the User's Guide.

SIMP-6667 #close

### CODE REVIEW

File: docs/help/FAQ/10_Password_Complexity.rst [hunk 1]
```diff
@@ -17,17 +17,29 @@ The default complexity is enforced in both :term:`PAM` and :term:`LDAP`.
...
+  * No more than 3 repetitions of a character from the same character class
+    * OK: ``abc``
+    * BAD: ``abcd``
+
   * No more than 4 characters in a monotonic character sequence
+    * OK: ``1b2c3d``
+    * BAD: ``aBcDe``
+
+  * Cannot contain your username
+  * Cannot contain items from your GECOS field (usually your full name)
```
Comment (lnemsick-simp) [inline docs/help/FAQ/10_Password_Complexity.rst]: pam module description says 'straight or reversed form'.  Should we indicate that detail here?

File: docs/help/FAQ/10_Password_Complexity.rst [hunk 2]
```diff
@@ -17,17 +17,29 @@ The default complexity is enforced in both :term:`PAM` and :term:`LDAP`.
...
+    * OK: ``aab``
+    * BAD: ``aaa``
+
+  * No more than 3 repetitions of a character from the same character class
+    * OK: ``abc``
+    * BAD: ``abcd``
+
   * No more than 4 characters in a monotonic character sequence
+    * OK: ``1b2c3d``
+    * BAD: ``aBcDe``
```
Comment (lnemsick-simp) [inline docs/help/FAQ/10_Password_Complexity.rst]: man page shown '12345' and 'fedcb' as examples.  Do we want a descending sequence as an example?

File: docs/help/FAQ/10_Password_Complexity.rst [hunk 3]
```diff
@@ -17,17 +17,29 @@ The default complexity is enforced in both :term:`PAM` and :term:`LDAP`.
...
   * 1 Upper case letter
   * 1 Lower case letter
   * 1 Number
   * 1 Special character
   * No more than 2 repetitions of the same character
+    * OK: ``aab``
+    * BAD: ``aaa``
+
+  * No more than 3 repetitions of a character from the same character class
+    * OK: ``abc``
```
Comment (lnemsick-simp) [inline docs/help/FAQ/10_Password_Complexity.rst]: Might be helpfuls if the example shows a switch from one character class to another on the 4th character.  For example ``abcD``

File: docs/help/FAQ/10_Password_Complexity.rst [hunk 4]
```diff
@@ -17,17 +17,32 @@ The default complexity is enforced in both :term:`PAM` and :term:`LDAP`.
...
   * 1 Number
   * 1 Special character
   * No more than 2 repetitions of the same character
+  
+    * OK: ``aab``
+    * BAD: ``aaa``
+
+  * No more than 3 repetitions of a character from the same character class
+  
+    * OK: ``abc``
```
Comment (op-ct) [inline docs/help/FAQ/10_Password_Complexity.rst]:
```suggestion
    * OK: ``abcD``
```

File: docs/help/FAQ/10_Password_Complexity.rst [hunk 5]
```diff
@@ -17,17 +17,32 @@ The default complexity is enforced in both :term:`PAM` and :term:`LDAP`.
...
+    * BAD: ``abcd``
+
   * No more than 4 characters in a monotonic character sequence
+  
+    * OK: ``1b2c3d``
+    * BAD: ``aBcDe``
```
Comment (op-ct) [inline docs/help/FAQ/10_Password_Complexity.rst]: I just tested against the pam module's defaults, using the acceptance tests:
  * The example ``aBcDe`` fails (as expected) against el6/`cracklib`, but **passes** against el7/`pw_quailty`.
  * A descending sequence of >4 chars like ``FeDcB`` fails against el6/`cracklib`, but **passes** against el7/`pw_quailty`.

File: docs/help/FAQ/10_Password_Complexity.rst [hunk 6]
```diff
@@ -17,17 +17,32 @@ The default complexity is enforced in both :term:`PAM` and :term:`LDAP`.
...
+
+  * Cannot contain your username
+  * Cannot contain items from your GECOS field (usually your full name)
+  * Must have more than 4 character changes from the old password
```
Comment (op-ct) [inline docs/help/FAQ/10_Password_Complexity.rst]: I just tested this against the pam module's defaults (from a pry session in the acceptance tests).  The policy enforces the documentation for normal users changing their own passwords, but the root user can change a normal user's password by only a single character using either el6/`cracklib` and el7/`pw_quailty`.

**Comment by lnemsick-simp** [comment]
Do we want to tell users on el7 they can use pwscore to check their password?

File: docs/help/FAQ/10_Password_Complexity.rst (hunk 6 above)
Comment (trevor-vaughan) [reply docs/help/FAQ/10_Password_Complexity.rst]: Well, sure. `root` can also edit `/etc/shadow` directly so I don't think we need to tell people that `root` has magic powers.

File: docs/help/FAQ/10_Password_Complexity.rst (hunk 5 above)
Comment (trevor-vaughan) [reply docs/help/FAQ/10_Password_Complexity.rst]: :man_facepalming: I think I'm going to note that behavior may differ slightly between `cracklib` and `pw_quality`. The man page says that it still works the same way so it's probably just a bug.

**Pull request merged**